content/terraform-plugin-framework/v1.15.x/docs/plugin/framework/resources/identity.mdx

```diff
@@ -0,0 +1,462 @@
+
+Practitioners can create an resource identity schema, like the existing managed resource schema, that consists of a map of attribute names and associated behaviours. With a few key differences:
+
+1. The identity schema will only support primitive types and list types such as:
+    - `BoolAttribute`
+    - `Float32Attribute`
+    - `Float64Attribute`
+    - `Int32Attribute`
+    - `Int64Attribute`
+    - `NumberAttribute`
+    - `StringAttribute`
+    - `ListAttribute`
+        - `ElementType: types.BoolType`
+        - `ElementType: types.Float32Type`
+        - `ElementType: types.Float64Type`
+        - `ElementType: types.Int32Type`
+        - `ElementType: types.Int64Type`
+        - `ElementType: types.NumberType`
+        - `ElementType: types.StringType`
+
+NOTE: The ordering in lists is expected to be stable and semantically relevant.
+
+2. Unlike the resource config schema, the resource identity schema will not support behaviours such as Required and Computed. Two new, but similar, behaviours will be allowed for each attribute: `RequiredForImport` and `OptionalForImport`.
+ - `RequiredForImport` is set to true if an import block would need this information as part of the identity attribute to import the resource. Some attributes might be specified through other means e.g. the provider configuration and therefore won't be needed for import.
+
+  - `OptionalForImport` is set to true if an import block can provide this information as part of the identity attribute to import the resource. This can be data that would normally be the default of the configured provider running the import. This does not mean the data is omitted from the resource identity, import-only data is not supported.
+
+It is expected that exactly one of `RequiredForImport` or `OptionalForImport` is set to true.
+
+3. Identity attributes are expected to be unique, immutable and fully known.
+
+Example of an identity schema for an aws_s3_bucket:
+```go
+var _ resource.ResourceWithIdentity = &S3BucketResource{} // new interface
+
+func (r *S3BucketResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_s3_bucket"
+}
+
+func (r *S3BucketResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// .. existing resource schema for aws_s3_bucket
+	}
+}
+
+func (r *S3BucketResource) IdentitySchema(_ context.Context, _ resource.IdentitySchemaRequest, resp *resource.IdentitySchemaResponse) {
+	resp.IdentitySchema = identityschema.Schema{
+		Attributes: map[string]identityschema.Attribute{
+			"bucket": identityschema.StringAttribute{
+				RequiredForImport: true,
+			},
+			"account_id": identityschema.StringAttribute{
+				OptionalForImport: true,
+			},
+			"partition": identityschema.StringAttribute{
+				OptionalForImport: true,
+			},
+			"region": identityschema.StringAttribute{
+				OptionalForImport: true,
+
+			},
+		},
+	}
+}
+```
```

Comment:
This is a good start, before we can do anything, we must choose the attributes for an identity, step 1!